llvm/lib/Target/AIE/AIEPostPipeliner.h

```diff
@@ -51,6 +51,14 @@ class NodeInfo {
   // The latest cycle at which this can be scheduled. This is a negative value
   // relative to the length of the linear schedule.
   int Latest = -1;
+
+  // Record critical path components
+  // The Pred/Succ that pushed my Earliest/Latest
+  int LastEarliestPusher = -1;
+  int LastLatestPusher = -1;
```

Comment:
Those should be `std::optional` :)

Reply:
Also, why do we compute the "last" pusher, and not the "worst"/"most critical" pusher?

Reply:
Hmm, looking at the code, I think we do find the "worst" pushers.

Reply:
By design, yes. Would a name change help? Suggestions will be given serious consideration.

Reply:
I guess I have a slight preference for `Worst` yes, but I don't really mind. Just having `EarliestPusher` also looks good to me. I can't find a better name at least.
Also, could this be a `NodeInfo *EarliestPusher = nullptr` instead of `std::optional<int>`?